File: Web/eUniversity.Web/Scripts/viewModels/curriculum/headerSectionViewModel.js
function HeaderSectionViewModel(serverModel) {
    var self = this;
    self.IsActive = ko.observable(true);
    self.IsCompleted = ko.observable(false);
    self.SpecializationId = ko.observable();
    self.CountSemesters = ko.observable();
    self.DateOfEnactmentDisplay = ko.observable();
    
    ko.mapping.fromJS(serverModel, {}, self);
}

File: Business/eUniversity.Business.ViewModels/Curriculum/HeaderSectionViewModel.cs
﻿using System;

namespace eUniversity.Business.ViewModels.Curriculum
{
    /// <summary>
    /// View model for curriculum header servition
    /// </summary>
    public class HeaderSectionViewModel
    {
        public long Id { get; set; }
        public long? SpecializationId { get; set; }
        public SemesterEnum CountSemesters { get; set; }

        public DateTime DateOfEnactment
        {
            get
            {
                return !string.IsNullOrEmpty(DateOfEnactmentDisplay) ? DateTime.Parse(DateOfEnactmentDisplay) : DateTime.Today;
            }
            set
            {
                DateOfEnactmentDisplay = value.ToShortDateString();
            }
        }

        public string DateOfEnactmentDisplay { get; set; }
    }
}